### PR TITLE
Support vs2019

### PR DESCRIPTION
**Pull request by 13xforever**

### CODE REVIEW

**Comment by 13xforever** [comment]
Before:
```
MSBUILDS:
14.0 C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe
12.0 C:\Program Files (x86)\MSBuild\12.0\Bin\MSBuild.exe
v4.0.30319 C:\WINDOWS\Microsoft.NET\Framework64\v4.0.30319\msbuild.exe
v3.5 C:\WINDOWS\Microsoft.NET\Framework64\v3.5\msbuild.exe
v2.0.50727 C:\WINDOWS\Microsoft.NET\Framework64\v2.0.50727\msbuild.exe
v4.0.30319 C:\WINDOWS\Microsoft.NET\Framework\v4.0.30319\msbuild.exe
v3.5 C:\WINDOWS\Microsoft.NET\Framework\v3.5\msbuild.exe
v2.0.50727 C:\WINDOWS\Microsoft.NET\Framework\v2.0.50727\msbuild.exe 
```

After:
```
MSBUILDS:
16.0.450.56488 C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Current\Bin\MSBuild.exe
14.0.25420.1 C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe
12.0.40629.0 C:\Program Files (x86)\MSBuild\12.0\Bin\MSBuild.exe
4.7.3190.0 C:\WINDOWS\Microsoft.NET\Framework64\v4.0.30319\msbuild.exe
3.5.30729.9034 C:\WINDOWS\Microsoft.NET\Framework64\v3.5\msbuild.exe
2.0.50727.9031 C:\WINDOWS\Microsoft.NET\Framework64\v2.0.50727\msbuild.exe
4.7.3190.0 C:\WINDOWS\Microsoft.NET\Framework\v4.0.30319\msbuild.exe
3.5.30729.9034 C:\WINDOWS\Microsoft.NET\Framework\v3.5\msbuild.exe
2.0.50727.9031 C:\WINDOWS\Microsoft.NET\Framework\v2.0.50727\msbuild.exe 
```

File: Common/Helper.cs [hunk 1]
```diff
@@ -335,5 +342,45 @@ public static string WindowsPathSlashesToUnix(string path)
...
         }
+
+        public static string GetMsBuildVersion(string fullPathToMsBuild)
+        {
+            if (!File.Exists(fullPathToMsBuild))
+                return null;
+
+            try
+            {
+                using (var proc = new Process
```
Comment (KungA) [inline Common/Helper.cs]: Prefer to use Cement ShellRunner instead raw Process.

File: Common/VsDevHelper.cs [hunk 2]
```diff
@@ -85,11 +85,12 @@ private static string FindVsDevCmd()
             if (programFiles == null)
                 return null;
 
-            foreach (var version in Helper.VisualStudioVersions())
+            foreach (var version in Helper.VisualStudioVersions)
+            foreach (var edition in Helper.VisualStudioEditions)
             {
                 paths.Add(new KeyValuePair<string, string>(
                     "VS150COMNTOOLS",
```
Comment (KungA) [inline Common/VsDevHelper.cs]: Does vs2019 have such variable with greater number, like `VSxxxCOMNTOOLS`?

Comment (13xforever) [reply Common/VsDevHelper.cs]:
```
D:\Develop\cement
λ set | grep -i comntools
VS110COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio 11.0\Common7\Tools\
VS120COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\Tools\
VS140COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\
VS160COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\Tools\
```
but only when you run a Developer Command Prompt

**Pull request merged**